From: =?utf-8?b?RnLDqWTDqXJpYyBCb25uYXJk?= <frediz@debian.org>
Date: Mon, 22 Feb 2021 09:14:58 +0100
Subject: Additions to apparmor profiles
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

o Debian uses dash as its default shell
o Debian has it's git components under /usr/lig/git-core/
o /usr/bin/git is being called directly when scm=git
o unzip is called just 'unzip' on Debian
o git takes a look at /etc/mailname for creating commit messages
  (perhaps it wouldn't if the Author were explicitly set, so maybe this
  is not really needed?)
o git-remote-{http*,rm} get used for pushing changes to needles,
  if enabled
o kvm uses seabios to boot
o needles need to be writable to enable editing
o pwd is in /bin on Debian
o git wants to look at /etc/gitconfig
o need to access OVMF files for UEFI booting VMs

Author: Philip Hands <phil@hands.com>,
        Frédéric Bonnard <frediz@debian.org>
---
 profiles/apparmor.d/usr.share.openqa.script.openqa | 27 +++++++++++++++-------
 profiles/apparmor.d/usr.share.openqa.script.worker | 18 ++++++++++-----
 2 files changed, 31 insertions(+), 14 deletions(-)

diff --git a/profiles/apparmor.d/usr.share.openqa.script.openqa b/profiles/apparmor.d/usr.share.openqa.script.openqa
index 8af6d6c..acbe857 100644
--- a/profiles/apparmor.d/usr.share.openqa.script.openqa
+++ b/profiles/apparmor.d/usr.share.openqa.script.openqa
@@ -16,23 +16,31 @@
   #include <abstractions/nameservice>
   #include <abstractions/perl>
 
+  /etc/gitconfig r,
   /etc/openqa/database.ini r,
   /etc/openqa/openqa.ini r,
   /etc/openqa/templates/ r,
   /etc/openqa/templates/** r,
+  /etc/mailname r,
   /proc/meminfo r,
   /tmp/** rwk,
+  /usr/bin/dash ix,
   /usr/bin/perl ix,
   /usr/bin/ssh rcx,
   /usr/bin/timeout rix,
-  /usr/bin/unzip-plain rix,
-  /usr/lib/git/git rix,
-  /usr/lib/git/git-add rix,
-  /usr/lib/git/git-commit rix,
-  /usr/lib/git/git-push rix,
-  /usr/lib/git/git-rebase rix,
-  /usr/lib/git/git-remote rix,
-  /usr/lib/git/git-rm rix,
+  /usr/bin/unzip{,-plain} rix,
+  /usr/bin/git rix,
+  /usr/bin/git-lfs rix,
+  /usr/lib/git{,-core}/git rix,
+  /usr/lib/git{,-core}/git-add rix,
+  /usr/lib/git{,-core}/git-commit rix,
+  /usr/lib/git{,-core}/git-push rix,
+  /usr/lib/git{,-core}/git-rebase rix,
+  /usr/lib/git{,-core}/git-remote rix,
+  /usr/lib/git{,-core}/git-remote-http rix,
+  /usr/lib/git{,-core}/git-remote-https rix,
+  /usr/lib/git{,-core}/git-rm rix,
+  /usr/lib/git{,-core}/git-upload-pack rix,
   /usr/libexec/git/git rix,
   /usr/libexec/git/git-add rix,
   /usr/libexec/git/git-commit rix,
@@ -74,6 +82,7 @@
   /var/lib/openqa/share/tests/ r,
   /var/lib/openqa/share/tests/** rw,
   /var/lib/openqa/share/tests/*/needles/.git/objects/*/* rwl,
+  /var/lib/openqa/share/tests/*/.git/hooks/* rix,
   /var/lib/openqa/testresults/ r,
   /var/lib/openqa/testresults/** rw,
   /var/lib/openqa/webui/** rw,
@@ -97,6 +106,8 @@
 
   }
 
+  /etc/ssl/openssl.cnf r,
+
   # Site-specific additions and overrides. See local/README for details.
   #include <local/usr.share.openqa.script.openqa>
 }
diff --git a/profiles/apparmor.d/usr.share.openqa.script.worker b/profiles/apparmor.d/usr.share.openqa.script.worker
index 94d7bc6..6ca70ef 100644
--- a/profiles/apparmor.d/usr.share.openqa.script.worker
+++ b/profiles/apparmor.d/usr.share.openqa.script.worker
@@ -26,7 +26,7 @@
 
   network netbeui raw,
 
-  /{usr/,}bin/bash rix,
+  /{usr/,}bin/{b,d}ash rix,
   /{usr/,}bin/pkill rix,
   /dev/ r,
   /dev/bus/usb/ r,
@@ -36,6 +36,7 @@
   /dev/ptmx rw,
   /dev/pts/* rw,
   /dev/tty rw,
+  /etc/gitconfig r,
   /etc/libnl/classid r,
   /etc/openqa/client.conf r,
   /etc/openqa/workers.ini r,
@@ -53,6 +54,7 @@
   /proc/meminfo r,
   /proc/sys/vm/overcommit_memory r,
   /proc/sys/fs/pipe-max-size r,
+  /proc/sys/vm/nr_hugepages r,
   /run/nscd/group r,
   /run/openqa/vde.ctl/* rw,
   /run/udev/queue.bin r,
@@ -64,9 +66,10 @@
   /sys/devices/system/node/** r,
   /sys/firmware/devicetree/base/ r,
   /sys/fs/cgroup/systemd/openqa.slice/** rw,
+  /sys/kernel/mm/transparent_hugepage/enabled r,
   /tmp/* rwk,
   /usr/bin/Xvnc rCx,
-  /usr/bin/cat rix,
+  /{usr/,}bin/cat rix,
   /usr/bin/chattr rix,
   /usr/bin/cksum rix,
   /usr/bin/cp rix,
@@ -88,7 +91,7 @@
   /usr/bin/mkdir rix,
   /usr/bin/nice rix,
   /usr/bin/optipng rix,
-  /usr/bin/pwd rix,
+  /{usr/,}bin/pwd rix,
   /usr/bin/python3.* ix,
   /usr/bin/qemu-img rix,
   /usr/bin/qemu-kvm rix,
@@ -112,8 +115,8 @@
   /usr/lib*/qemu/block-iscsi.so mr,
   /usr/lib*/qemu/block-rbd.so mr,
   /usr/lib*/qemu/block-ssh.so mr,
-  /usr/lib/git/git rix,
-  /usr/lib/git/git-remote-http rix,
+  /usr/lib/git{,-core}/git rix,
+  /usr/lib/git{,-core}/git-remote-http rix,
   /usr/lib/os-autoinst/videoencoder rix,
   /usr/libexec/git/git rix,
   /usr/libexec/git/git-remote-http rix,
@@ -127,7 +130,9 @@
   /usr/share/git-core/templates/ r,
   /usr/share/git-core/templates/** r,
   /usr/share/qemu/* rk,
+  /usr/share/seabios/* r,
   /usr/share/qemu/keymaps/* r,
+  /usr/share/OVMF/* rk,
   /var/lib/openqa/cache/ r,
   /var/lib/openqa/cache/** rwk,
   /var/lib/openqa/pool/ r,
@@ -143,9 +148,10 @@
   /var/lib/openqa/share/factory/tmp/ rw,
   /var/lib/openqa/share/factory/tmp/** rw,
   /var/lib/openqa/share/tests/** r,
+  /var/lib/openqa/share/tests/*/needles/** rw,
+  /var/lib/openqa/share/tests/*/profiles/*/needles/** rw,
   owner /sys/**/ rw,
 
-
   profile /usr/bin/Xvnc {
     #include <abstractions/X>
     #include <abstractions/base>
